### PR TITLE
Parseable JSON and text output in quiet mode for `dbt show` and `dbt compile`

### LINKED ISSUE #9840: [Feature] Allow `dbt show` to output json data without extra logs (by updating `--quiet`)
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

I would like the output of `dbt show` to be valid JSON, containing just the data and no logs.

Some potential solutions:
- `dbt show` could change its behaviour for `--quiet`, printing the results.
- or we could have a new parameter to `dbt show` that performs this behaviour if we don't want to change the `--quiet` behaviour

What I want to achieve is to pipe a `dbt show ... --output json` into `jq` or `jless`. Today` dbt --quiet show ...` doesn't print anything to stdout, which feels counterintuitive (is it a bug actually?)

### Describe alternatives you've considered

As suggested in the Community Slack I tried `dbt --log-format json show ... --output=json` ; but while the data is valid JSON, it doesn't show the results in a nice way as the real data results are encoded in a single string with escaped characters. 

### Who will this benefit?

People using the CLI and wanting a better/usable way to explorer the results of `dbt show`.

### Are you interested in contributing this feature?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dbeatty10**
resolves https://github.com/dbt-labs/dbt-core/issues/9840 (along with https://github.com/dbt-labs/dbt-common/pull/216)

### Problem

As described in #9840:
> I would like the output of `dbt show` to be valid JSON, containing just the data and no logs.

`dbt list` uses the `--quiet` flag to isolate the desired output from the log output, allowing the results to be piped or redirected.

But currently, `dbt show` and `dbt compile` do not work similarly. Rather, the `--quiet` flag suppresses _all_ output.

### Solution

The latest solution uses https://github.com/dbt-labs/dbt-common/pull/216 so that `CompiledNode` and `ShowNode` keep their same event names in the JSON logs, but also allow it to be emitted without timestamps even when `--quiet`.

When `--quiet`, also skips any extraneous output like:
- `Previewing node 'my_model':`
- `Previewing inline node:`
- `Compiled node 'my_model' is:`
- `Compiled inline node is:`

### 🎩

```shell
$ dbt show --inline "select 1 as id" -q --output json | jq .   

{
  "show": [
    {
      "id": 1
    }
  ]
}
```

```shell
$ dbt compile --inline "select 1 as id" -q --output json | jq .

{
  "compiled": "select 1 as id"
}
```

```shell
$ dbt show --select my_model --log-format json --quiet
$ cat logs/dbt.log

...
{"data": {"is_inline": false, "node_name": "my_model", "output_format": "text", "preview": "| id |\n| -- |\n|  1 |\n", "unique_id": "model.my_project.my_model"}, "info": {"category": "", "code": "Q041", "extra": {}, "invocation_id": "52649b96-c807-4ccc-b3dc-516882c15104", "level": "info", "msg": "Previewing node 'my_model':\n| id |\n| -- |\n|  1 |\n", "name": "ShowNode", "pid": 19995, "thread": "MainThread", "ts": "2024-11-19T01:45:49.800578Z"}}
...
```

```shell
$ dbt compile --select my_model --log-format json --quiet
$ cat logs/dbt.log

...
{"data": {"compiled": "select 1 as id", "is_inline": false, "node_name": "my_model", "output_format": "text", "unique_id": "model.my_project.my_model"}, "info": {"category": "", "code": "Q042", "extra": {}, "invocation_id": "d46ba24d-5968-4384-9277-e460600570ec", "level": "info", "msg": "Compiled node 'my_model' is:\nselect 1 as id", "name": "CompiledNode", "pid": 21075, "thread": "MainThread", "ts": "2024-11-19T01:47:07.203861Z"}}
...
```

<details>
<summary>

### Initial Solution (not adopted)
</summary>

The initial solution adopted the same exact approach as `dbt list` [here](https://github.com/dbt-labs/dbt-core/blob/4c1d0e92cd0a9c09659cc432c016dbc966b89daf/core/dbt/task/list.py#L178-L182), and basically copy-pasted from there.

Similar to how https://github.com/dbt-labs/dbt-core/pull/10131 stopped using `ListCmdOut` in favor of `PrintEvent`, this PR stopped using `CompiledNode` and `ShowNode` in favor of `PrintEvent`.

👉 So any consumers relying on `CompiledNode` or `ShowNode` existing with JSON logs wouldn't see those anymore but would see only `PrintEvent` instead. So I switched to the latest solution, to avoid any unintentional breakage for anyone creating and parsing JSON logs for `dbt show` / `dbt compile`.

### Before vs. after

Scenarios:
- `show` vs. `compile`
- `--select` vs. `--inline`
- `--quiet` vs. `--no-quiet`
- `--output text` vs. `json`
- `--log-format text` vs. `json`

#### Example for initial solution

```shell
dbt show --select my_model --log-format json --quiet
```

`logs/dbt.log` before:
```json
{"data": {"is_inline": false, "node_name": "my_model", "output_format": "text", "preview": "| event_id |   date_day |\n| -------- | ---------- |\n|        1 | 2002-02-02 |\n", "unique_id": "model.my_project.my_model"}, "info": {"category": "", "code": "Q041", "extra": {}, "invocation_id": "b295d589-99ba-450f-b261-0d60c5bcc195", "level": "info", "msg": "Previewing node 'my_model':\n| event_id |   date_day |\n| -------- | ---------- |\n|        1 | 2002-02-02 |\n", "name": "ShowNode", "pid": 96009, "thread": "MainThread", "ts": "2024-11-04T19:40:52.452943Z"}}
```

`logs/dbt.log` after:
```json
{"data": {"msg": "| event_id |   date_day |\n| -------- | ---------- |\n|        1 | 2002-02-02 |\n"}, "info": {"category": "", "code": "Z052", "extra": {}, "invocation_id": "d0857f0c-b58b-4374-893c-c454aba291b5", "level": "info", "msg": "| event_id |   date_day |\n| -------- | ---------- |\n|        1 | 2002-02-02 |\n", "name": "PrintEvent", "pid": 96415, "thread": "MainThread", "ts": "2024-11-04T19:41:43.897976Z"}}
```

This shows a difference in the JSON messages in the logs if `CompiledNode` and `ShowNode` events are converted to `PrintEvent` events.

In contrast, the solution we decided upon keeps `CompiledNode` and `ShowNode` events so the logs stay the same and don't change.

</details>

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests
- [x] This PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: core/dbt/task/compile.py [hunk 1]
```diff
@@ -89,16 +90,22 @@ def task_end_messages(self, results):
...
+                output_format=output_format,
+                unique_id=result.node.unique_id,
+                quiet=get_flags().QUIET,
             )
 
+            if get_flags().LOG_FORMAT == "json":
+                fire_event(compiled_node_event)
+            else:
+                # Cleaner to leave as print than to mutate the logger not to print timestamps.
+                print(compiled_node_event.message())
```
Comment (ChenyuLInx) [inline core/dbt/task/compile.py:107]: We can use the new PrintEvent being added to dbt-common.

Comment (dbeatty10) [reply core/dbt/task/compile.py:107]: Awesome! I'll check out https://github.com/dbt-labs/dbt-common/pull/130 and give `PrintEvent` a try.

Comment (dbeatty10) [reply core/dbt/task/compile.py:107]: Note to self: see https://github.com/dbt-labs/dbt-core/pull/10131 for an example how to use `PrintEvent`.

**Comment by dbeatty10** [comment]
@b-per FYI this and https://github.com/dbt-labs/dbt-common/pull/216 are meant to solve your request in https://github.com/dbt-labs/dbt-core/issues/9840

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 git+https://github.com/dbt-labs/dbt-adapters.git@main
 git+https://github.com/dbt-labs/dbt-adapters.git@main#subdirectory=dbt-tests-adapter
-git+https://github.com/dbt-labs/dbt-common.git@main
+git+https://github.com/dbt-labs/dbt-common.git@dbeatty/9840-quiet-show
```
Comment (dbeatty10) [inline dev-requirements.txt:3]: Make sure to restore the original branch prior to final merge:
```suggestion
git+https://github.com/dbt-labs/dbt-common.git@main
```

File: .changes/unreleased/Features-20240416-123234.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: Allow `dbt show` and `dbt compile` to output JSON without extra logs
```
Comment (dbeatty10) [inline .changes/unreleased/Features-20240416-123234.yaml:2]:
```suggestion
body: Parseable JSON and text output in quiet mode for `dbt show` and `dbt compile`
```

**Pull request merged**